src/lib/sm.c

```diff
@@ -43,12 +54,14 @@ void sm_init(struct sm *m,
 
 	PRE(conf[state].flags & SM_INITIAL);
 
-	m->conf = conf;
-	m->state = state;
-	m->invariant = invariant;
-	m->is_locked = is_locked;
-	m->id = ++id;
-	m->pid = getpid();
+	*m = (struct sm){
```

Comment:
This looks cool as it's easier to grep out initialization with `grep "(struct sm){"`.